### ISSUE TITLE
ImportError in app frozen by py2exe 

### ISSUE THREAD

**Issue by marunguy** [jaraco/jaraco.classes#3]
When App frozen by py2exe is run, below error occurred.
Win 8.1 64bit, python 2.7.13 win32, jaraco.classes 1.4.3, py2exe 0.6.10a1
```
D:\test\dist>cqpocrdistsvr.exe
----------------------------------------
main : Unhandled error
Traceback (most recent call last):
  ...
  File "zipextimporter.pyo", line 74, in load_module
  File "cherrypy\__init__.pyo", line 66, in <module>
  File "zipextimporter.pyo", line 74, in load_module
  File "cherrypy\_cperror.pyo", line 128, in <module>
ImportError: No module named jaraco.classes.properties
```

This error is similar to [https://github.com/jaraco/jaraco.classes/issues/2](https://github.com/jaraco/jaraco.classes/issues/2)
if empty `__init__.py` is created in `jaraco` folder, my frozen app works well.

When `jaraco`, `jaraco.classes` or `jaraco.classes.properties` is added to `includes` or `packages` in setup.py for py2exe,
below error occurred.
```
D:\test>python setup_py2exe.py py2exe -c
running py2exe
...
*** searching for required modules ***
Traceback (most recent call last):
  File "setup_py2exe.py", line 112, in <module>
    ctypes_com_server=[]
  File "C:\devtool\py27x86\lib\distutils\core.py", line 151, in setup
    dist.run_commands()
  File "C:\devtool\py27x86\lib\distutils\dist.py", line 953, in run_commands
    self.run_command(cmd)
  File "C:\devtool\py27x86\lib\distutils\dist.py", line 972, in run_command
    cmd_obj.run()
  File "C:\devtool\py27x86\lib\site-packages\py2exe\build_exe.py", line 241, in run
    self._run()
  File "C:\devtool\py27x86\lib\site-packages\py2exe\build_exe.py", line 294, in _run
    self.find_needed_modules(mf, required_files, required_modules)
  File "C:\devtool\py27x86\lib\site-packages\py2exe\build_exe.py", line 1306, in find_needed_modules
    mf.import_hook(f)
  File "C:\devtool\py27x86\lib\site-packages\py2exe\mf.py", line 719, in import_hook
    return Base.import_hook(self,name,caller,fromlist,level)
  File "C:\devtool\py27x86\lib\site-packages\py2exe\mf.py", line 136, in import_hook
    q, tail = self.find_head_package(parent, name)
  File "C:\devtool\py27x86\lib\site-packages\py2exe\mf.py", line 204, in find_head_package
    raise ImportError, "No module named " + qname
ImportError: No module named jaraco
```

**Comment by jaraco** [comment]
This sounds like a bug with py2exe. Perhaps raise the issue with that project to see if they have any recommendations.

A search led me to [this question/answer](https://stackoverflow.com/questions/36191922/py2exe-error-namespace-packages-not-yet-supported-skipping-package-snowflake) where less than a year ago, py2exe was reporting an error that namespace packages weren't supported. Do you see that error in your output. Have you followed the project to see if they've added support for namespace packages since then?

There are many many projects out there that rely on namespace packages. I can't think of anything this project can do other than stop existing, but if you find something, don't hesitate to ping back.

**Comment by webknjaz** [comment]
Ref: https://github.com/jaraco/jaraco.util/blob/master/jaraco/__init__.py#L5-L8

**Comment by jaraco** [comment]
If py2exe is omitting the `__init__.py` file for the namespace package, it won't matter what we put in there. I can't find anything in the documentation about what's recommended to handle namespace packages.

I think I can probably dust off the [jaraco.office project](/jaraco/jaraco.office), in which I used to build executables of that project.

**Comment by jaraco** [comment]
I tried using py2exe to build a simple application and I ran into obstacle after obstacle.

The latest py2exe doesn't run on Python 3.6 and doesn't run on Python 2.7, so I had to install Python 3.4.

After doing that, I created this [minimal script](https://gist.github.com/d69427d72b89bc18b8f1858f68c5b4dc) for building an executable that does nothing more than import cherrypy.

I installed the latest py2exe and cherrypy into the Python 3.4 system environment and then ran the script, but it crashes with an [infinite recursion](https://gist.github.com/e376cb79577d8b701885abd03261661c).

So even without involving any namespace packages, py2exe is unstable and eroding. I'm not going to be able to solve this issue. If there are issues with namespace packages in py2exe, those issues need to be published and solved in py2exe. If there's something a namespace package needs to do to support namespace packages, that technique needs to be published (at least on StackOverflow and preferably with the project).

If I do try to [build an executable that simply imports jaraco.classes](https://gist.github.com/15a27d9853efcd8735a77679637b708f), that will [run but fail to find the classes module](https://gist.github.com/4f0278999f1662867f3b80d2217e80d6).

No amount of py2exe support in `jaraco.__init__` is going to help there, because pip removes that file when installing. Perhaps there's a solution installing with eggs, but those are deprecated also.

I just don't see anything that `jaraco.*` or any other namespace package can do at this time to fix py2exe. I'm open to any suggestions.
